### DISCUSSION TITLE
Can the parameter `-b BootstrappingValeCompiler` be ignored?

### DISCUSSION THREAD

**Discussion by Ashvith10** [ValeLang/Vale#629] [General] [answered]
I'm trying to write a Nix derivation to try out Vale for myself. I'm taking references from the shell script written for Ubuntu, and I've come across this prerequisites script with the "-b" flag taking in the value "BootstrappingValeCompiler", presumably for the older version of Vale compiler. Is this needed? Can I skip this? I don't want to deal with having to ELF-patch, as I've never done it before.

**Comment by Verdagon** [comment, marked as answer]
It probably is needed, as the vale compiler is partially self-hosting, it will need an old version of the vale compiler.

There *is* a way to run the compiler without the self-hosted parts, but it's quite in-depth and could be a bit tricky. I'm not sure if anyone but me can do that, and my schedule's pretty packed for the next few weeks. I'll leave this issue open because it would be nice to have a script to do just that, now that you mention that.

**Reply by Ashvith10** [reply]
@Verdagon I was able to use `autoPatchelfHook` to make the pre-built`valec` binaries for Ubuntu run on my system, but it looks like there's hardly any output.

**No arguments**
```console
./result/bin/valec
Must specify a command (build or help).
(panic)

```

**`build` as argument**
```console
./result/bin/valec build
Unable to find program java.
(panic)

```

**`help` as argument**
```console
./result/bin/valec help
<empty>
```

**`version` as argument**
```console
./result/bin/valec version
<empty again>
```
